CoursesManager.UI/ViewModels/Courses/CourseOverViewViewModel.cs

```diff
@@ -1,54 +1,108 @@
         {
+            _courseRepository = courseRepository;
+            _dialogService = dialogService;
+            _messageBroker = messageBroker;
+
             ChangeCourseCommand = new RelayCommand(ChangeCourse);
             DeleteCourseCommand = new RelayCommand(DeleteCourse);
             CurrentCourse = (Course)GlobalCache.Instance.Get("Opened Course");
-            Students = CurrentCourse.students;
-            ObservableCollection<Registration> registration = DummyDataGenerator.GenerateRegistrations(Students.Count, 1);
-            studentPayments = new ObservableCollection<CourseStudentPayment>();
+            Students = CurrentCourse.Students;
+
+            if (Students is not null)
+            {
+                ObservableCollection<Registration> registration = DummyDataGenerator.GenerateRegistrations(Students.Count, 1);
+                StudentPayments = new ObservableCollection<CourseStudentPayment>();
 
-                for (int i = 0 ; i < registration.Count - 1 ; i++)  {
-                CourseStudentPayment studentPayment = new CourseStudentPayment(Students[i], registration[i]);
-                studentPayments.Add(studentPayment);
+                for (int i = 0; i < registration.Count - 1; i++)
+                {
+                    CourseStudentPayment studentPayment = new CourseStudentPayment(Students[i], registration[i]);
+                    StudentPayments.Add(studentPayment);
                 }
+            }
         }
 
 
-        private void ChangeCourse()
+        private void DeleteCourse()
         {
-            
+
         }
 
-        private void DeleteCourse()
+        private async void ChangeCourse()
         {
 
+            await ExecuteWithOverlayAsync(async () =>
+            {
+                var dialogResult = await _dialogService.ShowDialogAsync<CourseDialogViewModel, Course>(CurrentCourse);
+
+                if (dialogResult.Outcome == DialogOutcome.Success)
+                {
+                    CurrentCourse = dialogResult.Data;
+                    
+                    
+                }
+            });
+        }
+
+        private async Task ExecuteWithOverlayAsync(Func<Task> action)
+        {
+            _messageBroker.Publish(new OverlayActivationMessage(true));
+            try
+            {
+                await action();
+            }
+            finally
+            {
+                _messageBroker.Publish(new OverlayActivationMessage(false));
+            }
         }
+
+        private System.Windows.Visibility isImageVisible;
+
+        public System.Windows.Visibility IsImageVisible { get => isImageVisible; set => SetProperty(ref isImageVisible, value); }
+
+        private System.Windows.Visibility isPlaceholderVisible;
+
+        public System.Windows.Visibility IsPlaceholderVisible { get => isPlaceholderVisible; set => SetProperty(ref isPlaceholderVisible, value); }
```

Comment:
Zou je dit met niet window gerelateerde dingen kunnen oplossen? Ik zie dat je iets met visibility doet. Dit is beter om te doen met een bool in combinatie met een van de beschikbare converters.